src/main/ruby/core/thread.rb

```diff
@@ -447,12 +447,7 @@ def inspect
              Truffle::ThreadOperations.set_thread_local(:$SAFE, value) }
 )
 
-Truffle::KernelOperations.define_hooked_variable(
-  :$!,
-  -> { Truffle::ThreadOperations.get_thread_local(:$!) },
-  -> value { Truffle::ThreadOperations.set_thread_local(:$!, value) }
-)
-
+Truffle::KernelOperations.define_read_only_global(:$!, -> { Truffle::ThreadOperations.get_thread_local(:$!) })
```

Comment:
So this is actually a bug fix, right?
Could you mention it in the commit message?

Reply:
Done.